pkg/controller/util.go

```diff
@@ -25,6 +25,39 @@ func sortPods(pods []*corev1.Pod) {
 	})
 }
 
+// podNames returns the names of input pods.
+func podNames(pods []*corev1.Pod) []string {
+	names := make([]string, 0, len(pods))
+	for _, pod := range pods {
+		names = append(names, pod.Name)
+	}
+	return names
+}
+
+// isPodRunningAndReady returns whether the input pod is running and ready.
+func isPodRunningAndReady(pod *corev1.Pod) bool {
+	// The pod phase must be "running".
+	if pod.Status.Phase != corev1.PodRunning {
+		return false
+	}
+
+	// It must not be in the terminating state.
+	if pod.DeletionTimestamp != nil {
+		return false
+	}
+
+	// All containers must be running and ready.
+	for i := len(pod.Status.ContainerStatuses) - 1; i >= 0; i-- {
+		container := pod.Status.ContainerStatuses[i]
```

Comment:
Is there any reason why we're traversing this reversed?

Reply:
No reason. I changed it. It was initially copied from `kubectl` (but logic there is more complicated cause it needs to find 1 "status reason" to show per pod).